From 7c918831a8aac463ffe3cdd7e39b36ad09eec34a Mon Sep 17 00:00:00 2001
From: Milan Crha <mcrha@redhat.com>
Date: Mon, 3 Nov 2014 14:04:39 +0100
Subject: [PATCH 0020/1800] [EMailBackend] Provide cancellable and show in UI
 prepare for a quit

There was no indication in the status bar what is going on when
the Evolution was quitting, neither the associated activity could
have missing cancellable. Providing the cancellable and showing
the activity in the Mail's status bar may help a bit.
---
 mail/e-mail-backend.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/mail/e-mail-backend.c b/mail/e-mail-backend.c
index d3113ba8cc..8e8c11bb5e 100644
--- a/mail/e-mail-backend.c
+++ b/mail/e-mail-backend.c
@@ -396,8 +396,14 @@ mail_backend_prepare_for_quit_cb (EShell *shell,
 		   camel_operation_cancel_all() had been called, but we want
 		   it alive for the following operations, thus reset it. */
 		g_cancellable_reset (cancellable);
+	} else {
+		cancellable = camel_operation_new ();
+		e_activity_set_cancellable (activity, cancellable);
+		g_object_unref (cancellable);
 	}
 
+	e_shell_backend_add_activity (E_SHELL_BACKEND (backend), activity);
+
 	list = camel_session_list_services (CAMEL_SESSION (session));
 
 	if (delete_junk) {
-- 
2.15.0

